### ISSUE TITLE
Add support for loading NPCs and creatures.

### ISSUE THREAD

**Issue by ColeDeanShepherd** [ColeDeanShepherd/TESUnity#9]

**Comment by demonixis** [comment]
Do you have some ideas to get started?

**Comment by ColeDeanShepherd** [comment]
Yeah, getting this working fully will be a little involved, though... The first step is to add support for new ESM Record types like CREA, BODY, NPC_, and LEVC. You can see the format of those records [here](http://www.mwmythicmods.com/argent/tech/es_format.html). You would add code to support this [here](https://github.com/ColeDeanShepherd/TESUnity/blob/master/Assets/Scripts/TES/ESMFile.cs). That file is a bit of a mess, and not well documented, so sorry for that. I might make a quick commit adding some comments explaining how to add support for new record types. You would have to make new record classes, add the records to [GetModelFilename](https://github.com/ColeDeanShepherd/TESUnity/blob/master/Assets/Scripts/TES/ESMFile.cs#L12), add the record types [here](https://github.com/ColeDeanShepherd/TESUnity/blob/master/Assets/Scripts/TES/ESMFile.cs#L1961) if needed. And you would maybe add a line of code [here](https://github.com/ColeDeanShepherd/TESUnity/blob/master/Assets/Scripts/TES/MorrowindEngine.cs#L494).

Then, depending on how extensively we want to support NPC's and creatures, we might have to edit [NIFObjectBuilder](https://github.com/ColeDeanShepherd/TESUnity/blob/master/Assets/Scripts/TES/NIFObjectBuilder.cs). One thing it doesn't support is skinned meshes, that's why the flag you looked at in your video was oriented strangely.

Wow. What a mess. This is what happens when you write prototype code without thinking much about a clean design, sorry about that. I've been meaning to refactor the entire project's source code, but I've only gotten around to refactoring most the code in the "Core" folder.

**Comment by demonixis** [comment]
Ah ah don't worry about the code and thanks for the explanation about the Record types. I'll try few things, now I know where to start!

**Comment by demonixis** [comment]
I did good progress on [my branch for creatures](https://github.com/demonixis/TESUnity/tree/feature-crea)

![image](https://cloud.githubusercontent.com/assets/1017126/19181905/5d4514b4-8c70-11e6-9557-08a04f80f8fa.png)

I have added blank files for PNC_, BODY and LEVC. I'll try to implement them asap.

**Comment by ColeDeanShepherd** [comment]
Very nice! I might clone your branch and get some work done this weekend.

**Comment by demonixis** [comment]
Great! I started to work on NPC too but it was not as easy as creatures, but as I already said, I'm new to Morrowind (for now 😎)

**Comment by ColeDeanShepherd** [comment]
Yeah, NPCs are going to be tricky, I think, because they use the character customization system.

**Comment by demonixis** [comment]
Yep... I have made some progress this morning on my side. I have discovered that the BODY record has more than on SubRecord, it contains MODL, NAME and FNAME but it's not in the doc.. Have you an idea?
